controllers/watermarkpodautoscaler_controller_test.go

```diff
@@ -307,37 +304,49 @@ func newRequest(ns, name string) reconcile.Request {
 	}
 }
 
-type fakeMetricsClient struct {
-	getExternalMetrics func(metricName string, namespace string, selector labels.Selector) ([]int64, time.Time, error)
-}
-
-//// GetResourceMetric gets the given resource metric (and an associated oldest timestamp)
-//// for all pods matching the specified selector in the given namespace
-func (f fakeMetricsClient) GetResourceMetric(resource corev1.ResourceName, namespace string, selector labels.Selector, container string) (metrics.PodMetricsInfo, time.Time, error) {
-	return nil, time.Time{}, nil
-}
-
-//
-//// GetRawMetric gets the given metric (and an associated oldest timestamp)
-//// for all pods matching the specified selector in the given namespace
-func (f fakeMetricsClient) GetRawMetric(metricName string, namespace string, selector labels.Selector, metricSelector labels.Selector) (metrics.PodMetricsInfo, time.Time, error) {
-	return nil, time.Time{}, nil
-}
-
-//
-//// GetObjectMetric gets the given metric (and an associated timestamp) for the given
-//// object in the given namespace
-func (f fakeMetricsClient) GetObjectMetric(metricName string, namespace string, objectRef *v2beta2.CrossVersionObjectReference, metricSelector labels.Selector) (int64, time.Time, error) {
-	return 0, time.Time{}, nil
+func addUpdateReactor(s *fakescale.FakeScaleClient) {
+	s.AddReactor("update", "deployments", func(rawAction testcore.Action) (handled bool, ret runtime.Object, err error) {
```

Comment:
For consistency between `addUpdateReactor` and `addGetReactor`, either put all reactor functions inline (update) or split (get). I have no strong preference between the two.